blackbox/docs/general/dql/joins.rst

```diff
@@ -201,17 +201,48 @@ Example with ``within`` scalar function::
     +--------------+---------+
     SELECT 2 rows in set (... sec)
 
-However, there are no specific optimisations for certain join conditions such
-as ``=`` (equals) yet. The JOIN operation is implemented as a generic nested
-loop that invokes the operation on every record of the left table with every
-record of the right table. However, these filters are applied in a distributed
-way for each shard of the used tables.
-
 Limitations
 ...........
 
  - Joining more than 2 tables can result in execution plans which perform
    poorly as there is no query optimizer in place yet.
 
+Available Join Algorithms
+-------------------------
+
+Nested-Loop Join Algorithm
+..........................
+
+The nested loop algorithm evaluates the join conditions on every record of the
+left table with every record of the right table in a distributed manner (for 
+each shard of the used tables). The right table is scanned once for every row
+in the left table.
+
+Hash-Join Algorithm
+...................
+
+The performance of `Equi-Joins`_  is substantially improved by using the
+`Hash Join`_ algorithm. The left relation is scanned and loaded into a hash 
+table using the attributes of the join conditions as hash keys. The right 
+relation is then scanned and the join condition values of every row are hashed
+and matched against the hash table we built for the left relation. The matching
+rows are then emmited as the results of the join operation.
```

Comment:
imho this is too much implementation doc. I'd prefer more general phasing like wikipedia is doing, e.g. `Once the hash table is built, scan the larger relation and find the relevant rows from the smaller relation by looking in the hash table.`
whats important and missing is that we use a block hash join and so are not limiting to load the left into memory at once